x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/job/process/ModelSizeStats.java

```diff
@@ -0,0 +1,302 @@
+        new ConstructingObjectParser<>(RESULT_TYPE_VALUE, true, a -> new Builder((String) a[0]));
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), Job.ID);
+        PARSER.declareLong(Builder::setModelBytes, MODEL_BYTES_FIELD);
+        PARSER.declareLong(Builder::setBucketAllocationFailuresCount, BUCKET_ALLOCATION_FAILURES_COUNT_FIELD);
+        PARSER.declareLong(Builder::setTotalByFieldCount, TOTAL_BY_FIELD_COUNT_FIELD);
+        PARSER.declareLong(Builder::setTotalOverFieldCount, TOTAL_OVER_FIELD_COUNT_FIELD);
+        PARSER.declareLong(Builder::setTotalPartitionFieldCount, TOTAL_PARTITION_FIELD_COUNT_FIELD);
+        PARSER.declareField(Builder::setLogTime,
+            (p) -> TimeUtil.parseTimeField(p, LOG_TIME_FIELD.getPreferredName()),
+            LOG_TIME_FIELD,
+            ValueType.VALUE);
+        PARSER.declareField(Builder::setTimestamp,
+            (p) -> TimeUtil.parseTimeField(p, TIMESTAMP_FIELD.getPreferredName()),
+            TIMESTAMP_FIELD,
+            ValueType.VALUE);
+        PARSER.declareField(Builder::setMemoryStatus, p -> MemoryStatus.fromString(p.text()), MEMORY_STATUS_FIELD, ValueType.STRING);
+    }
+
+    /**
+     * The status of the memory monitored by the ResourceMonitor. OK is default,
+     * SOFT_LIMIT means that the models have done some aggressive pruning to
+     * keep the memory below the limit, and HARD_LIMIT means that samples have
+     * been dropped
+     */
+    public enum MemoryStatus {
+        OK, SOFT_LIMIT, HARD_LIMIT;
+
+        public static MemoryStatus fromString(String statusName) {
+            return valueOf(statusName.trim().toUpperCase(Locale.ROOT));
+        }
+
+        @Override
+        public String toString() {
+            return name().toLowerCase(Locale.ROOT);
+        }
+    }
+
+    private final String jobId;
+    private final long modelBytes;
+    private final long totalByFieldCount;
+    private final long totalOverFieldCount;
+    private final long totalPartitionFieldCount;
+    private final long bucketAllocationFailuresCount;
+    private final MemoryStatus memoryStatus;
+    private final Date timestamp;
+    private final Date logTime;
+
+    private ModelSizeStats(String jobId, long modelBytes, long totalByFieldCount, long totalOverFieldCount,
+                           long totalPartitionFieldCount, long bucketAllocationFailuresCount, MemoryStatus memoryStatus,
+                           Date timestamp, Date logTime) {
+        this.jobId = jobId;
+        this.modelBytes = modelBytes;
+        this.totalByFieldCount = totalByFieldCount;
+        this.totalOverFieldCount = totalOverFieldCount;
+        this.totalPartitionFieldCount = totalPartitionFieldCount;
+        this.bucketAllocationFailuresCount = bucketAllocationFailuresCount;
+        this.memoryStatus = memoryStatus;
+        this.timestamp = timestamp;
+        this.logTime = logTime;
+    }
+
+    public String getId() {
```

Comment:
Can `getId()` and `documentIdPrefix()` be removed? They are used in persistence only